.github/workflows/update-semgrep-rules.yml

```diff
@@ -0,0 +1,59 @@
+name: Update Semgrep Rules
+on:
+  workflow_dispatch: {}
+  # TODO: Remove this trigger
+  push:
+    branches:
+      - "cron"
+  #TODO: at some point also set up an actual cron like
+  #  schedule:
+  #  - cron: "43 20 * * *"
+
+jobs:
+  update-semgrep-rules-submodule:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get JWT for semgrep-ci GitHub App
+        id: jwt
+        uses: docker://public.ecr.aws/y9k7q4m1/devops/cicd:latest
+        env:
+          EXPIRATION: 600 # seconds
+          ISSUER: ${{ secrets.SEMGREP_CI_APP_ID }} # semgrep-ci GitHub App id
+          PRIVATE_KEY: ${{ secrets.SEMGREP_CI_APP_KEY }}
+      - name: Get token for semgrep-ci GitHub App
+        id: token
+        run: |
+          TOKEN="$(curl -X POST \
+          -H "Authorization: Bearer ${{ steps.jwt.outputs.jwt }}" \
+          -H "Accept: application/vnd.github.v3+json" \
+          "https://api.github.com/app/installations/${{ secrets.SEMGREP_CI_APP_INSTALLATION_ID }}/access_tokens" | \
+          jq -r .token)"
+          echo "::add-mask::$TOKEN"
+          echo "::set-output name=token::$TOKEN"
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: "recursive"
+          ref: "${{ github.event.repository.default_branch }}"
```

Comment:
can you explain the goal of this magic incantation? Which problem does it solve?

Reply:
why can't we simply use git checkout develop?

Reply:
Using git checkout develop would require some form of configuring URLs or auth using the token - this handles that for us.

Reply:
`develop` may not be the name of the default branch in the repository. It could be `master`, `main`, or any arbitrary other name configured by the repository owner. Using indirection here ensures this workflow continue to run even if the default branch name is changed in the future.

Reply:
I understand it's better to use default_branch, but I still don't understand why 'git checkout develop' fails if run after.